packages/docusaurus-plugin-content-docs/src/sidebars.js

```diff
@@ -113,19 +113,10 @@ function normalizeSidebar(sidebars) {
   }, {});
 }
 
-module.exports = function loadSidebars({siteDir, env}, deleteCache = true) {
```

Comment:
We can't do "hot reload" for the sidebars because we no longer are in charge of loading the sidebar. Not sure what's a good way around this. Possible solutions:

1. Tell users to use [`import-fresh`](https://github.com/sindresorhus/import-fresh) in `docusaurus.config.js` instead of Node's default `require`s if they want the hot reloading. Better ideas are welcome!
1. Parse the requires needed in `docusaurus.config.js` (possibly traversing the dependency graph too?) and bust the `require.cache` for the imports.
1. Ask users to provide a path to the sidebars file so that we do the loading for them. This is the closest approach to the old version but I don't really like it.

Reply:
I think one of the reason it doesn't do hot reload is because now the 'sidebars.json' is in fact not added to the chokidar watch. Not being able to trigger `load`.

I don't mind 3rd approach as long as we still have the hot reload functionality. Otherwise it's quite a regression 😭 . It's less migration too. 

Currently with this PR, user can do either
```js
sidebar: require('./sidebars.json'),
```
or putting the actual object inside docusaurus.config.js
```js
"sidebar": "docs": {
    "Getting Started": [
      "installation",
      "site-preparation",
      "site-creation",
      "publishing",
      "docker"
    ],
    "Guides": [
      "adding-blog",
      "custom-pages",
      "search",
      "navigation",
      "translation",
      "versioning"
    ],
    "API": [
      "commands",
      "doc-markdown",
      "api-pages",
      "site-config"
    ]
  }
```

But remember that we used to talk about bloating siteConfig, this can even make our config incredibly long. Better separation concern as a JSON is better though.

Reply:
Even Gatsby plugin won't do require inside gatsby.config.js. 
As far as I know, they will use require.resolve which will give the path string but throw an error if that path doesnt exist. `Require` has quite a problem with caching

Example:
```js
module.exports = {
  plugins: [
    // Shortcut for adding plugins without options.
    "gatsby-plugin-react-helmet",
    {
      // Standard plugin with options example
      resolve: require.resolve(`/path/to/gatsby-local-plugin`),
    },
  ],
}
```

Reply:
I've changed it to `require.resolve` and load via `import-fresh` to non cache it.